### DISCUSSION TITLE
Database permissions broken due to `prisma reset`

### DISCUSSION THREAD

**Discussion by ydennisy** [supabase/supabase#5441] [Questions]
Hello!

I am unable to access my database using the service key, and getting the following error:

```
{
    hint: null,
    details: null,
    code: '42501',
    message: 'permission denied for schema public'
}
```

I found this [SO question](https://stackoverflow.com/questions/67551593/supabase-client-permission-denied-for-schema-public), which explains that I have deleted various permissions supabase creates when booting a new project. I tried the solution to restore permissions as per the answer but with no luck, I can now access the schema but not the tables.

So my question - is the only option I have at this stage is to delete my supabase project and start again? This is a bit of a headache as I would need to reset all the keys in various apps.

Thanks in advance!

**Comment by GaryAustin1** [comment]
Edit:  It is possible your SO link has the same information, in glancing at the info.  So below could be a repeat but is direct from Supabase devs.  Sorry.
This issue link (especially the linked code reference in Supabase) might help you restore the table permissions...
https://github.com/supabase/supabase/issues/3415

**Reply by ydennisy** [reply]
Thanks @GaryAustin1 I tried running those commands, but they granted access to the schema but not individual tables.

Would be nice to have this documented, or even better automated in some way through the UI or CLI.
